dbms/src/Flash/Mpp/AsyncRequestHandler.h

```diff
@@ -0,0 +1,408 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/Exception.h>
+#include <Common/FailPoint.h>
+#include <Flash/Mpp/GRPCCompletionQueuePool.h>
+#include <Flash/Mpp/GRPCReceiveQueue.h>
+#include <Flash/Mpp/MppVersion.h>
+#include <Flash/Mpp/ReceiverChannelTryWriter.h>
+#include <Flash/Mpp/TrackedMppDataPacket.h>
+#include <common/defines.h>
+#include <grpcpp/alarm.h>
+#include <grpcpp/completion_queue.h>
+
+#include <condition_variable>
+#include <mutex>
+
+namespace DB
+{
+namespace FailPoints
+{
+extern const char exception_when_construct_async_request_handler[];
+} // namespace FailPoints
+
+enum class AsyncRequestStage
+{
+    NEED_INIT,
+    WAIT_MAKE_READER,
+    WAIT_BATCH_READ,
+    WAIT_FINISH,
+    WAIT_RETRY,
+    WAIT_REWRITE,
+    FINISHED,
+};
+
+using SystemClock = std::chrono::system_clock;
+
+constexpr Int32 max_retry_times = 10;
+constexpr Int32 retry_interval_time = 1; // second
+constexpr Int32 batch_packet_count = 16;
```

Comment:
How about making this configurable?

Reply:
> How about making this configurable?

`batch_packet_count` has been set as configurable, other parameters are no need I think, as they are unlikely to be used.